lib/rspec/mocks/any_instance/stub_chain.rb

```diff
@@ -27,7 +27,6 @@ def create_message_expectation_on(instance)
 
         def invocation_order
           @invocation_order ||= {
-            :with => [nil],
             :and_return => [:with, nil],
             :and_raise => [:with, nil],
             :and_yield => [:with, :and_yield, nil],
```

Comment:
Are the `nil` values in these arrays still needed given your change below to use `fetch` with a fallback of `[nil]`.